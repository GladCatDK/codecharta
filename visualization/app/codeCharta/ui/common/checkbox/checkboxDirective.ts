"use strict";

/**
 * A simple checkbox customized for CodeCharta. Watches for model changes and calls the bound change-function
 */
class CheckboxDirective {

    private template: string = "<div class='input-field col s12'><input type='checkbox' ng-model='model' id='checkbox-directive-{{::$id}}'><label for='checkbox-directive-{{::$id}}'>{{::label}}</label></div>";
    private scope: any = {
        model: "=",
        label: "@",
        change: "&"
    };
    private restrict: string = "E";

    constructor() {}

    /**
     * Watches model changes and links the change function to it.
     */
    link(scope) {
        scope.$watch(
            ()=> {
                return scope.model;
            },
            ()=> {
                scope.change();
            }
        );
    }

}

export {CheckboxDirective};
