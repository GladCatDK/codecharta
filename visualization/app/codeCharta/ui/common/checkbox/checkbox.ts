import angular from "angular";
import {CheckboxDirective} from "./checkboxDirective.ts";

angular.module("app.codeCharta.ui.common.checkbox",[]);

angular.module("app.codeCharta.ui.common.checkbox").directive(
    "checkboxDirective",
    () => new CheckboxDirective()
);
