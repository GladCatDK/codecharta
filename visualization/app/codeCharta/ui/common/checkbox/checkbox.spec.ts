import {NGMock, NG} from "../../../../ng.mockhelper.ts";
import {IRootScopeService, ICompileService, IScope} from "angular";

import "./checkbox.ts";

describe("checkbox", ()=>{

    let $rootScope: IRootScopeService;
    let $compile: ICompileService;


    beforeEach(NGMock.mock.module("app.codeCharta.ui.common.checkbox"));

    beforeEach(NGMock.mock.inject((_$rootScope_, _$compile_) => {
        $rootScope = _$rootScope_;
        $compile = _$compile_;
    }));

    it("snapshot test", () => {
        let scope: IScope = $rootScope.$new();
        let element = $compile(NG.element('<checkbox-directive label="some label" model="model" change="change()"></checkbox-directive>'))(scope);
        scope.$apply();
        expect(element).toMatchSnapshot();
    });

});


