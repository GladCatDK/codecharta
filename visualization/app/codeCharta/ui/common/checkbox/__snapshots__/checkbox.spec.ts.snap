// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox snapshot test 1`] = `
Object {
  "0": <checkbox-directive
    change="change()"
    class="ng-scope ng-isolate-scope"
    label="some label"
    model="model"
  >
    <div
      class="input-field col s12"
    >
      <input
        class="ng-pristine ng-untouched ng-valid ng-empty"
        id="checkbox-directive-3"
        ng-model="model"
        type="checkbox"
      />
      <label
        class="ng-binding"
        for="checkbox-directive-3"
      >
        some label
      </label>
    </div>
  </checkbox-directive>,
  "length": 1,
}
`;
